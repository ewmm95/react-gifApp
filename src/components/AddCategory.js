import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( { setCategories } ) => {
    const [inputValue, setInputValue] = useState('');
    const handleInputChange = ({target: {value}}) => setInputValue(value)
    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length <= 2) return;
        setCategories( x => [inputValue, ...x] );
        setInputValue('');
    }
    return (
        <form onSubmit={handleSubmit}>
            <input
                className='input-search'
                type='text'
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}