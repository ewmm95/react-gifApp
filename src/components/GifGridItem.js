import React from 'react';
import 'animate.css';
import PropTypes from 'prop-types';

export const GifGridItem = ({title, url}) => {
    return (
        <div className='card animate__animated animate__zoomIn' tabIndex="0">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}