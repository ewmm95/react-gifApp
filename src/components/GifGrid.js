import React, { useState, useEffect } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import 'animate.css';

export const GifGrid  = ({category}) => {
    const { data: images, loading} = useFetchGifs(category);
    return (
        <>
            <h3 className='animate__animated animate__fadeIn'>{category}</h3>
            {loading && <h3  className='animate__animated animate__flash'>Cargando...</h3>}
            <div className='cards'>
                { images.map( img => <GifGridItem key={img.id} {...img}/>) }
            </div>
        </>
    )
}
