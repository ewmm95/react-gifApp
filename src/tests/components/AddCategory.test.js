import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import { AddCategory } from "../../components/AddCategory"

describe('Pruebas en componente <AddCategory/>', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });

    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Texto de prueba';
        input.simulate('change', { target: { value } } );
        expect(wrapper.find('input').prop('value')).toBe(value);
    });
    
    test('No debe postear la informacion en submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect(setCategories).not.toHaveBeenCalled();
    })
    
    test('Debe de llamar el set categories y limpiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'HunterXHunter';
        input.simulate('change', { target: { value } } );
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect(setCategories).toHaveBeenCalled();
        expect(wrapper.find('input').prop('value')).toBe('');
    })
    
})
