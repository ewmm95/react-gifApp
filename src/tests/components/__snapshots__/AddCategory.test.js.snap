// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <AddCategory/> debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="input-search"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
